### PR TITLE
fix(find-people-in-room): Wrong references and logic

### PR DESCRIPTION
**Pull request by reinzor**

### CODE REVIEW

**Review by LarsJanssenTUe — APPROVED** [review]

**Review by LarsJanssenTUe — APPROVED** [review]

**Review by LarsJanssenTUe — APPROVED** [review]
@MatthijsBurgh I don't know why this did not work before (or why filters were being use) but from I think this looks good (it also does not give any errors now ;P). If you have some thoughts after we merge we can discuss those next time.

File: robot_smach_states/src/robot_smach_states/human_interaction/find_people_in_room.py [hunk 1]
```diff
@@ -166,37 +168,41 @@ def execute(self, userdata=None):
             rospy.loginfo("{} people remaining after None-check".format(len(found_people)))
 
             robot_pose = self._robot.base.get_location()
-            # TODO: Check probable bug here
-            found_people = filter(lambda x: (x.pose.frame.p - robot_pose.frame.p).Norm() < self._look_distance,
-                                  found_people)
+            for z in range(len(list(found_people))):
+                y = list(found_people)[z]
```
Comment (MatthijsBurgh) [inline robot_smach_states/src/robot_smach_states/human_interaction/find_people_in_room.py:172]: Found people is already a list...

File: robot_smach_states/src/robot_smach_states/human_interaction/find_people_in_room.py [hunk 2]
```diff
@@ -166,37 +168,41 @@ def execute(self, userdata=None):
...
+                rospy.loginfo(" Position of found_people  is {}".format(y.pose))
+                if (y.pose.frame.p - robot_pose.frame.p).Norm() > self._look_distance:
+                    list(found_people).pop(z)
```
Comment (MatthijsBurgh) [inline robot_smach_states/src/robot_smach_states/human_interaction/find_people_in_room.py:175]: You created a list from a list and then you pop items. When you create a new list, it does do a deep-copy, empirically determined. So popping here is useless.

File: robot_smach_states/src/robot_smach_states/human_interaction/find_people_in_room.py [hunk 3]
```diff
@@ -166,37 +168,41 @@ def execute(self, userdata=None):
...
 
-            rospy.loginfo("{} people remaining after distance < {}-check".format(len(found_people), self._look_distance))
+            rospy.loginfo("{} people remaining after distance < {}-check".format(len(list(found_people)), self._look_distance))
```
Comment (MatthijsBurgh) [inline robot_smach_states/src/robot_smach_states/human_interaction/find_people_in_room.py:177]: Was already a list

File: robot_smach_states/src/robot_smach_states/human_interaction/find_people_in_room.py [hunk 4]
```diff
@@ -166,37 +168,41 @@ def execute(self, userdata=None):
...
 
             if self._properties:
                 for k, v in self._properties.items():
-                    found_people = filter(lambda x:
-                            self._check_person_property(x, k, v), found_people)
-                    rospy.loginfo("{} people remaining after {}={} check".format(len(found_people), k, v))
+                    found_people = [x for x in found_people if self._check_person_property(x, k, v)]
+                    rospy.loginfo("{} people remaining after {}={} check".format(len(list(found_people)), k, v))
```
Comment (MatthijsBurgh) [inline robot_smach_states/src/robot_smach_states/human_interaction/find_people_in_room.py:182]: Already a list

File: robot_smach_states/src/robot_smach_states/human_interaction/find_people_in_room.py [hunk 5]
```diff
@@ -166,37 +168,41 @@ def execute(self, userdata=None):
...
             result_people = []
 
             if self._query_entity_designator:
                 query_entity = self._query_entity_designator.resolve()
                 if query_entity:
-                    result_people = filter(lambda x: query_entity.in_volume(VectorStamped.from_framestamped(x.pose), 'in'),
-                                           found_people)
+                    result_people = found_people
+                    for z in range(len(list(found_people))):
+                        y = list(found_people)[z]
```
Comment (MatthijsBurgh) [inline robot_smach_states/src/robot_smach_states/human_interaction/find_people_in_room.py:191]: Already a list

File: robot_smach_states/src/robot_smach_states/human_interaction/find_people_in_room.py [hunk 6]
```diff
@@ -166,37 +168,41 @@ def execute(self, userdata=None):
...
+                        if not query_entity.in_volume(VectorStamped.from_framestamped(y.pose), 'in'):
+                            list(result_people).pop(z)
```
Comment (MatthijsBurgh) [inline robot_smach_states/src/robot_smach_states/human_interaction/find_people_in_room.py:193]: Popping from a deep-copied list is useless

**Review by MatthijsBurgh — COMMENTED** [review]
@reinzor you can do much better than this....

**Pull request merged**